lib/ansible/modules/network/cloudengine/ce_vxlan_global.py

```diff
@@ -0,0 +1,562 @@
+        exist = is_config_exist(self.config, cmd)
+        if exist:
+            self.existing["nvo3_ecmp_hash"] = "disable"
+        else:
+            self.existing["nvo3_ecmp_hash"] = "disable"
+
+    def get_end_state(self):
+        """get end state info"""
+
+        config = self.get_current_config()
+
+        self.end_state["bridge_domain"] = self.get_bd_list()
+
+        cmd = "ip tunnel mode vxlan"
+        exist = is_config_exist(config, cmd)
+        if exist:
+            self.end_state["tunnel_mode_vxlan"] = "enable"
+        else:
+            self.end_state["tunnel_mode_vxlan"] = "disable"
+
+        cmd = "assign forward nvo3 f-linecard compatibility enable"
+        exist = is_config_exist(config, cmd)
+        if exist:
+            self.end_state["nvo3_prevent_loops"] = "enable"
+        else:
+            self.end_state["nvo3_prevent_loops"] = "disable"
+
+        cmd = "assign forward nvo3 acl extend enable"
+        exist = is_config_exist(config, cmd)
+        if exist:
+            self.end_state["nvo3_acl_extend"] = "enable"
+        else:
+            self.end_state["nvo3_acl_extend"] = "disable"
+
+        self.end_state["nvo3_gw_enhanced"] = get_nvo3_gw_enhanced(config)
+
+        cmd = "assign forward nvo3 service extend enable"
+        exist = is_config_exist(config, cmd)
+        if exist:
+            self.end_state["nvo3_service_extend"] = "enable"
+        else:
+            self.end_state["nvo3_service_extend"] = "disable"
+
+        cmd = "assign forward nvo3 eth-trunk hash enable"
+        exist = is_config_exist(config, cmd)
+        if exist:
+            self.end_state["nvo3_eth_trunk_hash"] = "enable"
+        else:
+            self.end_state["nvo3_eth_trunk_hash"] = "disable"
+
+        cmd = "assign forward nvo3 ecmp hash enable"
+        exist = is_config_exist(config, cmd)
+        if exist:
+            self.end_state["nvo3_ecmp_hash"] = "enable"
+        else:
+            self.end_state["nvo3_ecmp_hash"] = "disable"
+
+    def work(self):
+        """worker"""
+
+        self.check_params()
+        self.config = self.get_current_config()
+        self.bd_info = self.get_bd_list()
+        self.get_existing()
+        self.get_proposed()
+
+        # deal present or absent
+        self.config_bridge_domain()
+        self.config_tunnel_mode()
+        self.config_assign_forward()
+        if self.commands:
+            self.cli_load_config(self.commands)
+            self.changed = True
+
+        self.get_end_state()
+        self.results['changed'] = self.changed
+        self.results['proposed'] = self.proposed
+        self.results['existing'] = self.existing
+        self.results['end_state'] = self.end_state
+        if self.changed:
+            self.results['updates'] = self.updates_cmd
+        else:
+            self.results['updates'] = list()
+
+        self.module.exit_json(**self.results)
+
+
+def main():
+    """Module main"""
+
+    argument_spec = dict(
```

Comment:
For integers please:
use `type='int`
remove the `isdigit` check from `check_params`